index.js

```diff
@@ -0,0 +1,251 @@
+      let listLength = Math.min(10, data.length);
+      let container = document.createElement("div");
+      container.style.height = `${74 * listLength}px`;
+      container.className = "container";
+      container.id = "container";
+      for (let i = 0; i < listLength; i++) {
+        let element = `${data[i].name}  (${data[i].symbol})`;
+        let aTag = document.createElement("a");
+        let para = document.createElement("p");
+        aTag.className = `aTag`;
+        aTag.href = `company.html?symbol=${data[i].symbol}`;
+        aTag.innerHTML = element;
+        para.appendChild(aTag);
+        container.appendChild(para);
+      }
+      results.appendChild(container);
+    }
+
+    function initSearch() {
+      let button = document.querySelector("button");
+      button.addEventListener("click", dataFetch);
+    }
+
+    initSearch();
+  }
+
+  function showLoadingScreen() {
+    let loading = document.createElement("div");
+    loading.innerHTML = `Loading...`;
+    let displayStock = document.getElementById("loadDiv");
+    displayStock.appendChild(loading);
+    return loading;
+  }
+
+  function displayCompanyPage() {
+    function catSymbol() {
+      let urlParams = new URLSearchParams(window.location.search);
+      queryString = urlParams.toString();
+      let symbol = queryString.split(`=`)[1];
+      return symbol;
+    }
+
+    async function fetchCompanyData() {
+      const loadingElement = showLoadingScreen();
+      let symbol = catSymbol();
+      let url = `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/company/profile/${symbol}`;
+      let response = await fetch(url);
+      let data = await response.json();
+      loadingElement.remove();
+      displayCompanyData(data);
+    }
+
+    function displayCompanyData(data) {
+      let stockDetails = document.getElementById("stock-details");
+      let dataStore = {
+        name: data.profile.companyName,
+        price: ` Stock Price: ${data.profile.price}$`,
+        description: data.profile.description,
+        percentage: `(+${Number(data.profile.changesPercentage).toFixed(3)} %)`,
+      };
+
+      let img = document.createElement("img");
+      img.src = data.profile.image;
+      img.className = "card-img";
+      stockDetails.appendChild(img);
+
+      let container = document.createElement("div");
+      for (const [key, value] of Object.entries(dataStore)) {
+        let div = document.createElement("div");
+        div.className = key;
+        div.innerText = value;
+        container.appendChild(div);
+      }
+      stockDetails.appendChild(container);
+
+      let percentage = document.querySelector(".percentage");
+      if (data.profile.changesPercentage < 0) {
+        let change = `(${percentage.innerHTML.slice(2)}`;
+        percentage.innerHTML = change;
+        percentage.classList.add("red");
+      } else {
+        percentage.classList.add("green");
+      }
+    }
+
+    function getDataForChart() {
```

Comment:
Don't mix between async await and .then() syntax in your code. choose one of the options and stay with it.